### ISSUE TITLE
Forcing the expiration of WAL archive files

### ISSUE THREAD

**Issue by julianzhang98** [pgbackrest/pgbackrest#787]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version:
2.15.1

2. PostgreSQL version:
11

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each:
CENTOS 7

4. Did you install pgBackRest from source or from a package?
source

5. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)
    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)
    - errors in the postgresql log file before or during the time you experienced the issue
    - log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)

7. Describe the issue:

Wondering if there's a way to only keep enough wal files to reach consistency for a backup.

Assuming that I'm super limited on disk, let's say that
--retention-full=1
--retention-archive=1

then every time I would run backup --type=full, pgbackrest would expire the old full backup and all WALs associated with it. 

However, if I did not backup for 10 days, then I would have 10 days of WAL remaining in the repo. Is there a configuration or way so that I can purge all the WALs and only retain enough WALs to reach a consistent state? I understand that this will make PITR impossible.

In other words, it says @ https://pgbackrest.org/1/configuration.html#section-expire/option-retention-archive that: "Note that the WAL segments required to make a backup consistent are always retained until the backup is expired regardless of how this option is configured." Is there a way to only retain only the WAL segments required to make a backup consistent for the last backup as well?

Thanks!

**Comment by julianzhang98** [comment]
for example, if setting retention-archive to 0 was allowed

**Comment by dwsteele** [comment]
Currently no -- this would require cooperation from the archive-push command to only enable archiving during backups.  Otherwise you would accumulate WAL until the next expire, which is more or less the case now.

This feature has been discussed but has not made it to the top of the queue.  Most of the relevant code has been migrated to C, but we would still need a way for the backup and archive push processes to communicate about their current status.
